examples/gcoap/gcoap_cli.c

```diff
@@ -144,13 +144,29 @@ static size_t _send(uint8_t *buf, size_t len, char *addr_str, char *port_str)
     sock_udp_ep_t remote;
 
     remote.family = AF_INET6;
-    remote.netif  = SOCK_ADDR_ANY_NETIF;
+
+    /* parse for interface */
+    int iface = ipv6_addr_split_iface(addr_str);
```

Comment:
The ping6 command allows this to be ignored if only one interface exist (using `#if GNRC_NETIF_NUMOF == 1U`). In that case the first (and only) interface is selected.

Reply:
I understand your request. Thinking of the issue in #8130 with dependence on GNRC, is it a problem that I'll need to use GNRC functions to solve the issue here? Or is that the purpose of #6413 (stack-independent netif), and we'll need to update this work when that PR is merged?

Reply:
> Or is that the purpose of #6413 (stack-independent netif), and we'll need to update the interface lookup here when that PR is merged?

Yes, but also: this application at the moment is also dependent on GNRC (it pulls in GNRC dependencies), so I think it is okay for now.